modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/uuid/ItUuidExpressionTest.java

```diff
@@ -65,6 +70,31 @@ public void testInvalidUuidStringInDynamicParams() {
         assertThat(t.getMessage(), containsString("Invalid UUID string"));
     }
 
+    /**
+     * {@code COALESCE} operator does not allow different types in its arguments.
+     */
+    @ParameterizedTest
+    @MethodSource("convertedFrom")
+    public void testCoalesceMissingTypesIsIllegal(TestTypeArguments arguments) {
+        IgniteException t = assertThrows(IgniteException.class, () -> {
```

Comment:
let's fix all places. It should be also 'assertThrowsSqlException' with error code check